internal/wrapper/wrap_handler.go

```diff
@@ -49,11 +49,8 @@ func WrapHandlerWithListeners(handler interface{}, listeners ...HandlerListener)
 		}
 		CurrentContext = ctx
 		result, err := callHandler(ctx, msg, handler)
-		if err != nil {
-			ctx = context.WithValue(ctx, "error", true)
```

Comment:
I'm curious, do you know why this wasn't working?

Reply:
Old behavior: If there was an error, add a boolean `true` to the context. Then, use that boolean to send an enhanced metric saying there was an error. But, don't do anything with the trace.

New behavior: If there was an error, pass that error directly to the functions that are called after the handler runs. Use the error both to send an enhanced metric *and* add the error to the trace span.

Reply:
The reason we did not add errors to the spans before was due to a bug in dd-trace-go. That bug has been fixed. So now we can add the functionality here. There was no bug here per se, we just hadn't implemented the functionality.

Reply:
Ok, awesome. Thanks!